## 6. 演習問題の解答
課題-2から課題-5までの解答をこの後に記述します．課題-5については，問題の意図に沿って全体に散りばめて記載してあります．
*****

### 課題-2: 簡単な計算
Jupyter notebookのCodeセルを使って，下記の計算を実施してください．ただし，問題ごとに個別のCodeセルに分けて実施してください．

最初の計算を行います．<br>
$12+34$

In [1]:
12+34

46

計算結果として46が得られました．次の問題に移ります．<br>
$77-99$

In [2]:
77-99

-22

計算結果として-22が得られました．次の問題に移ります．<br>
$63\times 7$

In [3]:
63*7

441

計算結果として441が得られました．次の問題に移ります．<br>
$82\div 6$

In [4]:
72/6

12.0

計算結果として12.0が得られました．次の問題に移ります．<br>
$3^4$

In [5]:
3**4

81

計算結果として81が得られました．次の問題に移ります．<br>
$\sqrt{5}$

In [6]:
5**(1/2)

2.23606797749979

計算結果として2.23606797749979が得られました．次の問題に移ります．<br>
$52\div 7$の割り切った値（実数）

In [7]:
52/5

10.4

計算結果として10.5が得られました．次の問題に移ります．<br>
$52\div 7$の商（整数）

In [8]:
52//5

10

計算結果として10が得られました．次の問題に移ります．<br>
$52\div 7$の余り（整数）

In [9]:
52%5

2

計算結果として10が得られました．
以上で課題-2が終わりました．

### 課題-3: 複雑な計算
ここでも計算を行いますが，若干複雑な式を計算します．ここでも，各問題を個別のCodeセルに分けて実施してください．

最初の計算を行います．<br>
$(4+11)\times (1.5-0.38)$

In [10]:
(4+11)*(1.5-0.38)

16.8

計算結果として16.8が得られました．次の問題に移ります．<br>
$\frac{18+7}{-3-7}$

In [11]:
(18+7)/(-3-7)

-2.5

計算結果として-2.5が得られました．次の問題に移ります．<br>
$3^{12}\div 3^9$

In [12]:
3**12 / 3**9

27.0

計算結果として27.0が得られました．次の問題に移ります．<br>
$\sqrt{1+2+3+4}$

In [13]:
(1+2+3+4)**(1/2)

3.1622776601683795

計算結果として3.1622776601683795が得られました．次の問題に移ります．<br>
$\yen 1,250+\yen 22,000$

In [14]:
1250 + 22000  # remove , from number format

23250

計算結果として23250が得られました．次の問題に移ります．<br>
$12億\div 百万$

In [15]:
1.2e9 / 1e6

1200.0

数字の桁数が大きいので計算に用いる数字を指数形式で活用します．
```
12億 = 1,200,000,000 = 1.2e9
百万 = 1,000,000 = 1e6
```
計算結果として1200.0が得られました．次の問題に移ります．<br>
1から5までの自然数を足して，合計値を5で割る

In [16]:
(1+2+3+4+5)/5

3.0

計算結果として3.0が得られました．以上で課題-3が終わりました．

### 課題-4: 変数を使った計算
この課題では，全ての問題で変数を使ってください．変数名については，意味がある場合は適切な名称を設定してください．意味のない変数については$\,x\,$や$\,y\,$などを使用して構いません．

(1) $x=28$としたとき$x^2-5x+6$を計算せよ．<br>

In [17]:
x =28
x**2 - 5*x +6

650

この問題では，変数として$\,x\,$を定義しました．最初の式で$\,x\,$に28を代入し，2つ目の式で2次関数に$\,x\,$を代入しています．
計算結果として650が得られました．

次の問題に移ります．

(2) この問題は3つのCodeセルを使って実施せよ．<br>&nbsp;&nbsp;&nbsp;&nbsp;
    (a) 80円のトマトを5個，160年のキャベツを1個，40円のジャガイモを6個購入します．合計金額を計算せよ．<br>&nbsp;&nbsp;&nbsp;&nbsp;
    (b) 消費税を8%とした場合の支払金額を計算せよ．<br>&nbsp;&nbsp;&nbsp;&nbsp;
    (c) 消費税を10%とした場合の支払金額を計算せよ．

In [18]:
vegetables = 80*5 + 160*1 + 40*6
vegetables

800

In [19]:
vegetables * 1.08

864.0

In [20]:
vegetables * 1.10

880.0000000000001

最初のセルで野菜の合計金額を計算して，合計金額を変数vegetablesに代入しています．合計金額は800円でした．<br>
次のセルで消費税8%の場合の計算をしています.8%の消費税込みで864円となります．<br>
最後のセルでは消費税10%の場合の計算をしています.10%の消費税込みで880円となります．小数点以下は計算誤差です．<br>

次の問題に移ります．

(3) マラソンにおいて2時間20分45秒の記録で走りました．
    (a) この記録を秒数で示せ．
    (b) この記録を分数で示せ．

In [21]:
recordSeconds = 2*60*60 + 20*60 + 45
recordSeconds

8445

In [22]:
recordSeconds / 60

140.75

マラソンの記録2時間20分45秒を秒表示に変換しました．その値を変数recordSecondsに代入しました．記録は8445秒です．
recordSecondsを60秒で割ると分表示となります．分表示だと140.75分となります．
*****

以上が課題の解答となります．
これにて，**社会人のためのPython活用術入門**の履修を完了します．